src/controllers/api/2014/classController.ts

```diff
@@ -10,6 +11,21 @@ import SimpleController from '@/controllers/simpleController';
 import Spell from '@/models/2014/spell';
 import Subclass from '@/models/2014/subclass';
 
+// --- Zod Schemas ---
+const ParamsSchema = z.object({
+  index: z.string().min(1),
+});
+
+const LevelParamsSchema = z.object({
+  index: z.string().min(1),
+  level: z.coerce.number().int().min(1).max(20),
+});
+
+const LevelsQuerySchema = z.object({
+  subclass: z.string().min(1).optional(),
+});
```

Comment:
Does this not need to be `.optional()` as well?

e.g.
```ts
const LevelsQuerySchema = z.object({ ... }).optional()
```

I guess not as long as we know `req.query` is never `undefined`/`null` but instead is gonna be `{}` instead. Assuming thats true you can disregard this comment

Reply:
Yeah. Given it is now being properly validated, `validatedQuery.data` should always be at least `{}`. Which the controller logic handles correctly.